### DISCUSSION TITLE
Is it possible to generate the hyperlink outline list of a class member?

### DISCUSSION THREAD

**Discussion by rynnwu** [michaeljones/breathe#716] [Q&A]
Hi there, thanks for the awesome extension 👍 
If it possible to generate something like methods/attributes outline list as hyperlink at the beginning of class? For example:

source code which doxygen will read:
```
class MyClass {
 public:
  /** method1 */
  void method1();
 /** .method 2*/
  void method2();
  ...
  /** method10 */
  void method10();
};
```

And in reST, list a class using
```
.. doxygenclass:: MyClass
  :members:
```

What we got should be:

---
**classMyClass**
Public Functions

void method1()
*method1*

void method2()
*method2*

void method10()
*method10*

---

If the method counts is big, it's hard to see the overview of the methos.
I'd like to have a outline list like:

---
**classMyClass**
Public Functions List
- void method1()  <-- as hyperlink
- void method2()  <-- as hyperlink
- void method10()  <-- as hyperlink

Public Functions

void method1()
*method1*

void method2()
*method2*

void method10()
*method10*

---

Thanks.

**Comment by jakobandersen** [comment]
In short: no, not yet, but it's a good idea.
There are a couple of related issues, #667, #634, #378, #321, with the last one being similar to your suggestion, but yours seems more detailed in the description.
I think all the building blocks are in place to progress on this (i.e., the ``cpp:alias`` directive in Sphinx), but as it requires modification of the Breathe directives I'm reluctant to work on this before #698 is merged.

**Reply by rynnwu** [reply]
Got it and thanks :)

**Reply by vermeeren** [reply]
@jakobandersen As you mention this has a more detailed description, should we copypaste/ref it in one of the other ones? Or perhaps create an issue out of it so the info is not lost?

**Reply by jakobandersen** [reply]
Hmm, it may be good to create a new issue with a summary of what the linked ones ask for, so we can get a coherent view and discussion.
